### ISSUE TITLE
Allow login sessions longer than 1 week

### ISSUE THREAD

**Issue by probablykasper** [supabase/supabase#6713]
# Feature request

## Is your feature request related to a problem? Please describe.

Having to log in again and again is annoying, and I'm really glad most websites don't log me out after just a week. Supabase login sessions unfortunately seem to be limited to 1 week.

## Describe the solution you'd like

Ability to specify longer login sessions

**Comment by kiwicopple** [comment]
> Supabase login sessions unfortunately seem to be limited to 1 week

Hey @probablykasper - just a quick check here. Are you referring to our Dashboard, or to the Auth functions on your own project?

**Comment by probablykasper** [comment]
I'm referring to the `Authenitcation > Settings > General > JWT expiry` option, where it says this:
```
How long tokens are valid for, in seconds. Defaults to 3600 (1 hour), maximum 604,800 seconds (one week).
```

**Comment by gracebir** [comment]
@probablykasper , how long does the login session will last now ? I can raise it up to 1 month

**Comment by probablykasper** [comment]
I'd prefer 1 year

**Comment by gracebir** [comment]
@probablykasper , don't you think that 1 year will be longer for a session?

**Comment by probablykasper** [comment]
You mean too long? I don't think it's too long. Google, GitHub and many other websites seem to keep you logged in for a year

**Comment by kiwicopple** [comment]
Hey all, just to clarify, this value is how long the JWT is valid for until it needs to be reset. In theory, the session can last forever, as long as it can keep refreshing.

This should be _as low as possible_ for security reasons.

If you're having issues with your users getting logged out, then it's an issue with the refresh. We recently updated the Auth libraries to work better across multiple tabs - perhaps that was the issue? If so, can you please update supabase-js to the latest version and try again?

I'll close this issue now, because it is the correct behaviour, but feel free to comment below if you need more guidance (or open a Discusssion)

**Comment by probablykasper** [comment]
> Hey all, just to clarify, this value is how long the JWT is valid for until it needs to be reset. In theory, the session can last forever, as long as it can keep refreshing.

I see, thank you for explaining. I've updated the feature request to mention that Supabase limits login sessions to 1 week of absence. This is still way too short in my opinion - There are lots of websites I use regularly, but not weekly, and I'd rather not be logged out of those.

> This should be _as low as possible_ for security reasons.

Many sites keep me logged in way longer than 1 week after I visit the site. Do all those have security issues?

**Comment by kiwicopple** [comment]
After one week, you could still refresh the token (as long as you still have the `refresh_token`) it's just that the current short-lived JWT is no longer valid and it can't be used without refreshing it again.

Our code for refreshing the token is here:

https://github.com/supabase/gotrue-js/blob/30fb5729a1988dbd7f61d58ec10eb81de63f93c7/src/GoTrueClient.ts#L641

It looks like we remove the session _without_ attempting to refresh it after it has expired. I guess a case could be made for trying to refresh it once and only removing the session if there is a failure to refresh

**Comment by probablykasper** [comment]
Ah, I think I understand now. The JWT expiry should have nothing to do with whether you're logged in, the JWT is just a thing that avoids extra validation on the server. [This SO answer has a great explanation](https://stackoverflow.com/a/48525808/6553404).

It might be good to have an explanation of what JWT expiry means in the Supabase settings. I immediately maxed it out because I thought they were for login session lengths.

**Comment by t1u1** [comment]
> Hey all, just to clarify, this value is how long the JWT is valid for until it needs to be reset. In theory, the session can last forever, as long as it can keep refreshing.

Somewhat related question: Can two or more JWT tokens be issued at the same time, and can they be refreshed independently? This might happen, for example, when a user is logged in from two different devices, I assume two different JWT tokens will be created in this case.

**Comment by kiwicopple** [comment]
when you log in from two devices, you create 2 "sessions" in supabase - each has it's own `refresh_token`. These act independently

tldr: yes - you have different tokens for different devices (or browsers for that matter)
